### ISSUE TITLE
Is the following code for custom loss used in training correct?

### ISSUE THREAD

**Issue by biprajiman** [fchollet/keras#5229]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by biprajiman** [comment]
def prepare_model()
    #----pretrained model changed to functional api----    
    input_img = Input(shape=input_shape)
    dense=Dense(1024)(input_img)
    relu_dense=Activation('relu')(dense)
    softmax_dense=Activation('softmax')(dense)

    def custom_loss(y_true, y_pred):
        class_loss= K.categorical_crossentropy(y_true,y_pred)
        add_loss= K.mean(K.square(relu_dense - softmax_dense), axis=-1)
        return class_loss+lambda*add_loss

    model=Model(input=input_img, output=softmax_dense)

    model.compile(loss=custom_loss, optimizer='adadelta', metrics['accuracy'])

 return model
